packages/flutter/test/material/input_decorator_test.dart

```diff
@@ -759,6 +759,53 @@ void main() {
     expect(getBorderWeight(tester), 2.0);
   });
 
+  testWidgets('InputDecorator hint overflow', (WidgetTester tester) async {
+    final String longHint = 'hintText' * 10;
+
+    // Default Hint overflow is TextOverflow.ellipse
+    await tester.pumpWidget(
+      buildInputDecorator(
+        decoration: InputDecoration(
+          hintText: longHint,
+        ),
+      ),
+    );
+    await tester.pumpAndSettle();
+
+    Text text = tester.firstWidget(find.text(longHint));
+    expect(text.overflow, TextOverflow.ellipsis);
+
+    // Customizable hintOverflow
+    await tester.pumpWidget(
+      buildInputDecorator(
+        decoration: InputDecoration(
+          hintText: longHint,
+          hintOverflow: TextOverflow.fade,
+        ),
+      ),
+    );
+    await tester.pumpAndSettle();
+
+    text = tester.firstWidget(find.text(longHint));
+    expect(text.overflow, TextOverflow.fade);
+
+    // Customizable InputDecorationTheme hintOverflow
+    await tester.pumpWidget(
+      buildInputDecorator(
+        inputDecorationTheme: const InputDecorationTheme(
+          hintOverflow: TextOverflow.clip
+        ),
+        decoration: InputDecoration(
+          hintText: longHint,
+        ),
+      ),
+    );
+    await tester.pumpAndSettle();
+
+    text = tester.firstWidget(find.text(longHint));
+    expect(text.overflow, TextOverflow.clip);
```

Comment:
Add another one showing that InputDecoration(hintOverflow: something) takes precedence over InputDecorationTheme(hintOverflow: somethingElse).